src/main/java/com/salesforce/einsteinbot/sdk/client/BasicChatbotClientImpl.java

```diff
@@ -86,10 +97,23 @@ void setHealthApi(HealthApi healthApi) {
     this.healthApi = healthApi;
   }
 
+  @VisibleForTesting
+  void setVersionsApi(VersionsApi versionsApi) {
+    this.versionsApi = versionsApi;
+  }
+
   @Override
   public BotResponse startChatSession(RequestConfig config,
       ExternalSessionId sessionId,
       BotSendMessageRequest botSendMessageRequest) {
+    try {
+      if (!isApiVersionSupported()) {
+        String errorMessageFormat = "SDK failed to start chat as the API version is not supported. Current API version in SDK is %s, latest supported API version is %s, please upgrade to the latest version. Further information about specific versions is available here: https://github.com/forcedotcom/einstein-bot-sdk-java#version-numbering";
+        throw new RuntimeException(String.format(errorMessageFormat, getCurrentApiVersion(), getLatestApiVersion()));
+      }
+    } catch (IOException e) {
```

Comment:
Can you catch `IOException` in `getCurrentApiVersion` and rethrow as `RuntimeException` ? Also, let's create a separate exception like `UnsupportedSdkVersion` for this error.

Reply:
Makes sense, I will do that